lib/shared/config-manager/src/index.ts

```diff
@@ -67,26 +86,126 @@ export class EnvironmentConfigManager {
                 this.logger.debug('DevCycle initial config loaded')
             })
             .finally(() => {
-                if (this.disablePolling) {
-                    return
-                }
-                this.intervalTimeout = this.setInterval(async () => {
-                    try {
-                        await this._fetchConfig()
-                    } catch (ex) {
-                        this.logger.error((ex as Error).message)
-                    }
-                }, this.pollingIntervalMS)
+                this.startWatchingForConfigChanges()
             })
     }
 
-    stopPolling(): void {
-        this.disablePolling = true
+    startWatchingForConfigChanges(): void {
+        if (this.configFetchState === ConfigFetchState.DISABLED) {
+            this.logger.warn(
+                'Config fetching is disabled to start watching for config changes',
+            )
+            return
+        }
+
+        if (this.disableRealtimeUpdates) {
+            this.startPolling()
+        } else {
+            this.startSSE()
+        }
+    }
+
+    stopWatchingForConfigChanges(): void {
+        this.configFetchState = ConfigFetchState.DISABLED
+        this.stopPolling()
+        this.stopSSE()
+    }
+
+    private startSSE(): void {
+        if (this.configFetchState === ConfigFetchState.POLLING) {
+            this.stopPolling()
+        }
+
+        if (!this.configSSE) {
+            this.logger.warn('No SSE configuration found, switching to polling')
```

Comment:
what's this warning supposed to indicate to the user? and what would they do about it